tools/BuildAssets/psModules/CodeGenerationModules/generateDotNetSdkCode.psm1

```diff
@@ -0,0 +1,207 @@
+    }
+}
+
+function Write-ErrorLog {
+    param(
+        [string] $msg,
+        [switch] $logToConsole,
+        [switch] $logToFile
+    )
+    if(![string]::IsNullOrEmpty($msg) -and $logToFile)
+    {
+        $errorStream.Append("$msg`n") | Out-Null
+    }
+    if($logToConsole)
+    {
+        Write-Error $msg
+    }
+}
+
+function launchProcess {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $command, 
+        [string] $args)
+    $pinfo = New-Object System.Diagnostics.ProcessStartInfo
+    $pinfo.FileName = $command
+    $pinfo.RedirectStandardError = $true
+    $pinfo.RedirectStandardOutput = $true
+    $pinfo.UseShellExecute = $false
+    
+    if(![string]::IsNullOrEmpty($args))
+    {
+        $pinfo.Arguments = $args
+    }
+    
+    $p = New-Object System.Diagnostics.Process
+    $p.StartInfo = $pinfo
+    $p.Start() | Out-Null
+    $p.WaitForExit()
+    $stdout = $p.StandardOutput.ReadToEnd()
+    $stderr = $p.StandardError.ReadToEnd()
+    Write-InfoLog $stdout -logToConsole
+    Write-ErrorLog $stderr
+    if($p.ExitCode -ne 0)
+    {
+        throw [System.Exception] "Command $cmd $args returned $p.ExitCode"
+    }
+}
+
+function Get-AutoRestHelp {
+    Write-InfoLog "Fetching AutoRest help" -logToConsole
+    launchProcess "cmd.exe" "/c autorest.cmd --help"
+}
+
+function Install-AutoRest {
+    param(
+        [Parameter(Mandatory = $true)]
+        $AutoRestVersion
+    )
+    
+    Write-InfoLog "Installing AutoRest version: $AutoRestVersion" -logToConsole
+    
+    Try {
+        launchProcess "cmd.exe" "/c npm.cmd install -g autorest@$AutoRestVersion"
+    }
+    Catch [System.Exception] {
+        Write-ErrorLog $_.Exception.ToString()
+        throw [System.Exception] "AutoRest Installation failed"
+    }
+    Write-InfoLog "AutoRest installed successfully." -logToConsole
+}
+
+function Start-CodeGeneration {
+param(
+    [Parameter(Mandatory = $true)]
+    [string] $SpecsRepoFork,
+    [Parameter(Mandatory = $true)]
+    [string] $SpecsRepoBranch,
+    [Parameter(Mandatory = $true)]
+    [string] $SpecsRepoName,
+    [Parameter(Mandatory = $true, HelpMessage ="Please provide an output directory for the generated code")]
+    [string] $SdkDirectory,
+    [Parameter(Mandatory = $true, HelpMessage ="Please provide a version for the AutoRest release")]
+    [string] $AutoRestVersion
+    )
+    
+    $configFile="https://github.com/$SpecsRepoFork/$SpecsRepoName/blob/$SpecsRepoBranch/specification/$ResourceProvider/readme.md" 
+    Write-InfoLog "Generating CSharp code" -logToConsole
+    $cmd = "cmd.exe"
+    $args = "/c autorest.cmd $configFile --csharp --csharp-sdks-folder=$SdkDirectory --version=$AutoRestVersion --reflect-api-versions"
+    
+    Write-InfoLog "Executing AutoRest command" -logToFile
+    Write-InfoLog "$cmd $args" -logToFile
+
+    Try {
+        launchProcess $cmd $args
+    }
+    Catch [System.Exception] {
+        Write-ErrorLog $_.Exception.ToString()
```

Comment:
Will this cause errors to be logged twice, once here and then again where the bigger try catch might be